### PR TITLE
fix: error message rewrite when it has only a getter

### PR DESCRIPTION
**Pull request by atian25**
##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

### CODE REVIEW

File: lib/application.js [hunk 1]
```diff
@@ -164,7 +164,12 @@ class Application extends EggApplication {
       server: [ server ],
       error: (err, throwErrorCount) => {
         if (err.message) {
-          err.message += ' (uncaughtException throw ' + throwErrorCount + ' times on pid:' + process.pid + ')';
+          try {
+            err.message += ' (uncaughtException throw ' + throwErrorCount + ' times on pid:' + process.pid + ')';
+          } catch (_) {
+            // ignore, shouldjs will override error but missing setter.
+            // https://github.com/shouldjs/should.js/blob/889e22ebf19a06bc2747d24cf34b25cc00b37464/lib/assertion-error.js#L26
+          }
```
Comment (gxcsoccer) [inline lib/application.js]:
```js
const originMessage = err.message;

Object.defineProperty(this, "message", {
      get: function() {
        return originMessage + ' (uncaughtException throw ' + throwErrorCount + ' times on pid:' + process.pid + ')';
      },
      configurable: true,
      enumerable: false
    });
```

Comment (atian25) [reply lib/application.js]: 之前担心这样会影响到 shouldjs 的错误输出排版。

不过也可以吧，我改改

Comment (atian25) [reply lib/application.js]: 改了，@gxcsoccer

File: lib/application.js [hunk 2]
```diff
@@ -163,8 +163,17 @@ class Application extends EggApplication {
     graceful({
       server: [ server ],
       error: (err, throwErrorCount) => {
-        if (err.message) {
-          err.message += ' (uncaughtException throw ' + throwErrorCount + ' times on pid:' + process.pid + ')';
+        const originMessage = err.message;
```
Comment (gxcsoccer) [inline lib/application.js]:
```js
const originMessage = err.message || '';
```

Comment (atian25) [reply lib/application.js]: 下面有个 if，不会进去。

**Comment by fengmk2** [comment]
@atian25 ci 没过

**Pull request merged**